chainstate/tx-verifier/src/transaction_verifier/error.rs

```diff
@@ -143,10 +145,18 @@ pub enum ConnectTransactionError {
     DestinationRetrievalError(#[from] SignatureDestinationGetterError),
     #[error("Output timelock error: {0}")]
     OutputTimelockError(#[from] timelock_check::OutputMaturityError),
+    #[error("Nonce is not incremental: {0:?}")]
+    NonceIsNotIncremental(AccountType),
+    #[error("Nonce is not found: {0:?}")]
+    MissingTransactionNonce(AccountType),
     #[error(
         "Transaction {0} has not enough pledge to create a stake pool: giver {1:?}, required {2:?}"
     )]
     NotEnoughPledgeToCreateStakePool(Id<Transaction>, Amount, Amount),
+    #[error("Attempt to create stake pool from accounting inputs")]
+    AttemptToCreateStakePoolFromAccounts,
```

Comment:
More rules... more complexity.

I'll come back to this later when we're done with the VIU problem.